fundingmanager_test.go

```diff
@@ -0,0 +1,596 @@
+
+	// Alice respond with a FundingComplete messages
+	fundingComplete := <-aliceMsgChan
+	if fundingComplete.MsgType() != lnwire.MsgSingleFundingComplete {
+		t.Fatalf("expected MsgSingleFundingComplete to be sent from alice")
+	}
+
+	// give it to Bob
+	bobFundingMgr.processFundingComplete(fundingComplete.(*lnwire.SingleFundingComplete), aliceAddr)
+
+	// Finally, Bob should send the SingleFundingSignComplete message
+	fundinSignComplete := <-bobMsgChan
+	if fundinSignComplete.MsgType() != lnwire.MsgSingleFundingSignComplete {
+		t.Fatalf("expected MsgSingleFundingSignComplete to be sent from bob")
+	}
+
+	// forward signature to Alice
+	aliceFundingMgr.processFundingSignComplete(fundinSignComplete.(*lnwire.SingleFundingSignComplete), bobAddr)
+
+	// Small sleep to make sure the transaction is broadcast
+	time.Sleep(100 * time.Millisecond)
+
+	// After Alice processes the singleFundingSignComplete message, she will
+	// broadcast the funding transaction to the network. This means that it
+	// should be part of the next block (in our test environment)
+	blockHashes, err := miningNode.Node.Generate(1)
+	if err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	block, err := miningNode.Node.GetBlock(blockHashes[0])
+	if err != nil {
+		t.Fatalf("unable to get block: %v", err)
+	}
+
+	// Block should contain coinbase tx + our funding tx
+	if len(block.Transactions) != 2 {
+		t.Fatalf("expected transaction to be part of block")
+	}
+
+	// TODO: Check that this transaction == the funding transaction
+	var fundingTxHash chainhash.Hash
+	for _, transaction := range block.Transactions {
+		fundingTxHash = transaction.TxHash()
+	}
+
+	fundingOutPoint := &wire.OutPoint{
+		Hash:  fundingTxHash,
+		Index: 0,
+	}
+
+	// Give fundingManager time to process the newly mined tx
+	time.Sleep(100 * time.Millisecond)
+
+	// The funding transaction was mined, so assert that both funding managers
+	// now have the state of this channel 'markedOpen' in their internal state
+	// machine.
+	state, _, err := aliceFundingMgr.getChannelOpeningState(fundingOutPoint)
+	if err != nil {
+		t.Fatalf("unable to get channel state: %v", err)
+	}
+
+	if state != markedOpen {
+		t.Fatalf("expected state to be markedOpen, was %v", state)
+	}
+	state, _, err = bobFundingMgr.getChannelOpeningState(fundingOutPoint)
+	if err != nil {
+		t.Fatalf("unable to get channel state: %v", err)
+	}
+
+	if state != markedOpen {
+		t.Fatalf("expected state to be markedOpen, was %v", state)
+	}
+
+	// After the funding transaction is mined, Alice will send fundingLocked to Bob
+	fundingLockedAlice := <-aliceMsgChan
+	if fundingLockedAlice.MsgType() != lnwire.MsgFundingLocked {
+		t.Fatalf("expected fundingLocked sent from Alice")
+	}
+
+	// And similarly Bob will send funding locked to Alice
+	fundingLockedBob := <-bobMsgChan
+	if fundingLockedBob.MsgType() != lnwire.MsgFundingLocked {
+		t.Fatalf("expected fundingLocked sent from Bob")
+	}
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Ideally, we can eliminate this sleep. I think that we can use the `ArbiterChan` to synchronize this exchange instead.